src/main/webapp/app/entities/feedback.model.ts

```diff
@@ -114,6 +147,36 @@ export class Feedback implements BaseEntity {
         return Feedback.hasDetailText(that) || !!(that.gradingInstruction && that.gradingInstruction.feedback);
     }
 
+    /**
+     * Get the referenced file path for referenced programming feedbacks, or undefined.
+     * Typical reference format for programming feedback: `file:src/com/example/package/MyClass.java_line:13`.
+     * Example output in this case: `src/com/example/package/MyClass.java`
+     */
+    public static getReferenceFilePath(feedback: Feedback): string | undefined {
+        if (!feedback.reference?.startsWith('file:')) {
+            // No programming feedback
+            return undefined;
+        }
+        return feedback.reference.substring(5).split('_')[0];
+    }
+
+    /**
+     * Get the referenced line for referenced programming feedbacks, or undefined.
+     * Typical reference format for programming feedback: `file:src/com/example/package/MyClass.java_line:13`.
+     * Example output in this case: 13
+     */
+    public static getReferenceLine(feedback: Feedback): number | undefined {
+        if (!feedback?.reference?.startsWith('file:')) {
+            // No programming feedback
+            return undefined;
+        }
+        const line = parseInt(feedback.reference.split('_line:')[1]);
```

Comment:
We are handling unreferenced feedback 🤔  `getReferenceLine` should never be called for unreferenced feedback. Even if it is, the first line of this function (`if (!feedback?.reference?.startsWith('file:')) {` would be true, and `getReferenceLine` would return `undefined`. Where do you see the potential for an error?

Reply:
I mean in my module I'm referencing a filename: `src/com/example/package/MyClass.java` without line start and line end 😬 I guess for right now you'd be returning `null` for the reference field from Athena instead of `file:src/com/example/package/MyClass.java`. I guess it does not make much sense to have unreferenced feedback in Artemis that references a file 🤔

Reply:
Ok! I suggest we keep the code as-is in both your module and Artemis and add a compatibility filter within Artemis as a follow-up PR.

This could detect unreferenced file feedback and change it to "pure" unreferenced feedback where the `text` starts with `(file name):`. So when the module suggests `{reference: "src/com/example/package/MyClass.java", detailText: "Hello world"}`, it would be converted to `{reference: undefined, detailText: "src/com/example/package/MyClass.java: Hello world"}`

What do you think?

Reply:
Sounds good, will anyways have to see if my programming module is any good 🤔